src/main/java/com/google/api/codegen/discovery/config/ApiaryConfigToSampleConfigConverter.java

```diff
@@ -86,53 +88,95 @@ public SampleConfig convert() {
   /** Creates a method. */
   private MethodInfo createMethod(Method method) {
     // The order of fields must be preserved, so we use an ImmutableMap.
-    ImmutableMap.Builder<String, FieldInfo> fields = new ImmutableMap.Builder<>();
+    ImmutableMap.Builder<String, FieldInfo> fieldsBuilder = new ImmutableMap.Builder<>();
     TypeInfo requestBodyType = null;
+    Type requestType = apiaryConfig.getType(method.getRequestTypeUrl());
     for (String fieldName : apiaryConfig.getMethodParams(method.getName())) {
-      Type containerType = apiaryConfig.getType(method.getRequestTypeUrl());
-      Field field = apiaryConfig.getField(containerType, fieldName);
+      Field field = apiaryConfig.getField(requestType, fieldName);
       // If one of the method arguments has the field name "request$", it's the
       // request body.
       if (fieldName.equals(DiscoveryImporter.REQUEST_FIELD_NAME)) {
         requestBodyType = createTypeInfo(field, method);
         continue;
       }
-      fields.put(field.getName(), createFieldInfo(field, containerType, method));
+      fieldsBuilder.put(field.getName(), createFieldInfo(field, requestType, method));
     }
+    ImmutableMap<String, FieldInfo> fields = fieldsBuilder.build();
 
-    TypeInfo requestType = createTypeInfo(method, true);
-    TypeInfo responseType = null;
+    TypeInfo requestTypeInfo = createTypeInfo(method, true);
+    TypeInfo responseTypeInfo = null;
     String responseTypeUrl = typeNameGenerator.getResponseTypeUrl(method.getResponseTypeUrl());
     if (!Strings.isNullOrEmpty(responseTypeUrl)) {
-      responseType = createTypeInfo(method, false);
+      responseTypeInfo = createTypeInfo(method, false);
     }
 
-    boolean isPageStreaming = isPageStreaming(method);
-    FieldInfo pageStreamingResourceField = null;
-    if (isPageStreaming) {
-      Type containerType = apiaryConfig.getType(responseTypeUrl);
-      Field field = getPageStreamingResourceField(containerType);
-      pageStreamingResourceField = createFieldInfo(field, containerType, method);
-    }
     boolean isPageStreamingResourceSetterInRequestBody = false;
+    String requestPageTokenName = "";
     if (requestBodyType != null) {
-      isPageStreamingResourceSetterInRequestBody =
-          requestBodyType.message().fields().containsKey(PAGE_TOKEN_FIELD_NAME);
+      Map<String, FieldInfo> requestBodyFields = requestBodyType.message().fields();
+      for (String tokenName : PAGE_TOKEN_NAMES) {
```

Comment:
For this and the loop below, we could simplify a bit by writing a function `String hasField(MessageType messageType, String name...);` that returns the name of the field if found and `null` otherwise